### PR TITLE
Cleanups, fixes, documentation and tests for wlgen

### PR DESCRIPTION
**Pull request by bjackman**
- Provides some cleanups and minor fixes for the `wlgen` package.
- Adds some docstrings describing the API.
- Adds very simple tests for the `RTA` and `PerfMessaging` classes, using a LocalLinuxTarget to avoid requiring a connected external target. This approach has limitations because we don't want to give LISA root on a real machine (which is required e.g. for calibrating rt-app), but useful tests can be done that work around this limitation. Hopefully this could allow us to start self-testing parts of LISA in a CI loop.

### CODE REVIEW

**Comment by bjackman** [comment]
Ugh, Github messed up the commit order again, the commit order according to `git log` is:
```
cae0d87a * test_wlgen: Add smoke test for PerfMessaging (hackbench)
56411fb4 * test_wlgen: Make RTA tests actually run the workload
d16f8fb1 * Add smoke tests for wlgen RT-App classes
7c818d5d * submodules: Update devlib to latest master
0b1220b8 * wlgen: Add some Sphinx-compatible docstrings
cd63a54e * rta: Rename _TaskBase -> RTATask
5d59968f * rta: Create run_dir before pushing config JSON
c6897036 * perf_bench: Set up logging for PerfPipe
c62804a6 * perf_bench: Increase default loop count for PerfMessaging
82cb3980 * workload: Remove outdated comment
59666a46 * workload: Error if cgroup specified but no cgroups module
f62b3be7 * workload: Remove rt-app specific `calibration`
b2afbad2 * workload: Remove getTasks method
df85689f * workload: Remove unused cgroup_cmd instance variable
bc216397 * workload: Remove taskset instance variables
ebedb3c3 * workload: Formatting
```

File: libs/wlgen/wlgen/workload.py [hunk 1]
```diff
@@ -126,6 +126,7 @@ def conf(self,
         # Initialize run folder
         if self.run_dir is None:
             self.run_dir = self.target.working_directory
+        self.target.execute('mkdir -p {}'.format(self.run_dir))
```
Comment (bjackman) [inline libs/wlgen/wlgen/workload.py]: Yep, I knew about that function so no idea why I didn't use it.

File: libs/wlgen/wlgen/rta.py [hunk 2]
```diff
@@ -144,7 +144,7 @@ def __postrun(self, params):
             return
         self._log.debug('Pulling logfiles to [%s]...', destdir)
         for task in self.tasks.keys():
-            logfile = "'{0:s}/*{1:s}*.log'"\
+            logfile = "{0:s}/*{1:s}*.log"\
```
Comment (bjackman) [inline libs/wlgen/wlgen/rta.py]: Good point. Actually, I'll use `self.target.path` for the sake of theoretical consistency.

**Comment by bjackman** [comment]
OK, pushed a new version, commit log is now like:
```
b19a0338 test_wlgen: Add smoke tests for wlgen classes
351f346d wlgen: Add some Sphinx-compatible docstrings
7c57a58c rta: Use self.target.path instead of explicit '/'s
78f26862 rta: Don't put single quotes around glob patterns
ec260221 rta: Rename _TaskBase -> RTATask
6a5df67d perf_bench: Pass run_dir to superclass in conf
45af3315 perf_bench: Set up logging for PerfPipe
d822d25e perf_bench: Increase default loop count for PerfMessaging
0758a503 workload: Create run_dir in conf method
156fe82c workload: Remove outdated comment
c9af5a7b workload: Error if cgroup specified but no cgroups module
c2842a8a workload: Remove rt-app specific `calibration`
9f757767 workload: Remove getTasks method
629e77ed workload: Remove unused cgroup_cmd instance variable
6ee4d8e5 workload: Remove taskset instance variables
0ae35fb0 workload: Formatting
```
------
> I would suggest to split these last two in a separate PR for a better review while me can merge the cleanups

I can split out up to `156fe82c workload: Remove outdated comment` but after that there are dependencies. Do you think it's worth it?

File: libs/wlgen/wlgen/workload.py (hunk 1 above)
Comment (derkling) [inline libs/wlgen/wlgen/workload.py]: Is that still to be refreshed?

**Comment by derkling** [comment]
So... that's a pretty big PR and unfortunately github does not help too much on understanding which patches have been updated and in which order... :-/

I'm happy to merge all the "workload: " prefixed patches, provided you fixed the dirs creations... can you please send out a cleanup PR for just these patches and split further this one in:

PR1
```
6a5df67d perf_bench: Pass run_dir to superclass in conf
45af3315 perf_bench: Set up logging for PerfPipe
d822d25e perf_bench: Increase default loop count for PerfMessaging
```

PR2
```
7c57a58c rta: Use self.target.path instead of explicit '/'s
78f26862 rta: Don't put single quotes around glob patterns
ec260221 rta: Rename _TaskBase -> RTATask
```

PR3
```
b19a0338 test_wlgen: Add smoke tests for wlgen classes
351f346d wlgen: Add some Sphinx-compatible docstrings
```

You can send the as long as I merge the previous PR... but that would simplify a lot the review process.

**Comment by bjackman** [comment]
This has been split up and supersede by these PRs:

RTA cleanups: https://github.com/ARM-software/lisa/pull/296
perf_bench cleanups: https://github.com/ARM-software/lisa/pull/294
Workload cleeanups: https://github.com/ARM-software/lisa/pull/293
Documentation: https://github.com/ARM-software/lisa/pull/297
Self-tests: https://github.com/ARM-software/lisa/pull/298

**Pull request closed without merging**